langchain/chains/conversational_retrieval/base.py

```diff
@@ -188,13 +198,59 @@ def save(self, file_path: Union[Path, str]) -> None:
 
 
 class ConversationalRetrievalChain(BaseConversationalRetrievalChain):
-    """Chain for chatting with an index."""
+    """Chain for having a conversation based on retrieved documents.
+
+    This chain takes in chat history (a list of messages) and new questions,
+    and then returns an answer to that question.
+    The algorithm for this chain consists of three parts:
+
+    1. Use the chat history and the new question to create a "standalone question".
+    This is done so that this question can be passed into the retrieval step to fetch
+    relevant documents. If only the new question was passed in, then relevant context
+    may be lacking. If the whole conversation was passed into retrieval, there may
+    be unnecessary information there that would distract from retrieval.
+
+    2. This new question is passed to the retriever and relevant documents are
+    returned.
+
+    3. The retrieved documents are passed to an LLM along with the new question to
+    generate a final answer.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.chains import (
+                StuffDocumentsChain, LLMChain, ConversationalRetrievalChain
+            )
+            from langchain.prompts import PromptTemplate
+            from langchain.llms import OpenAI
+
+            combine_docs_chain = StuffDocumentsChain(...)
+            vectorstore = ...
```

Comment:
if we're providing working code in all other docstring examples should we do that here, too